fv3config/fv3run/_kubernetes.py

```diff
@@ -0,0 +1,266 @@
+            memory_gb (float, optional): gigabytes of memory required for the kubernetes
+                worker, defaults to 3.6GB
+            memory_gb_limit (float, optional): maximum memory allowed for the kubernetes
+                worker, defaults to the value set by memory_gb
+            cpu_count (int, optional): number of CPUs to use on the kubernetes worker
+            gcp_secret (str, optional): name of kubernetes secret to mount containing a
+                file ``key.json`` to use as the google cloud storage key.
+            image_pull_policy (str, optional): pull policy passed on to the
+                kubernetes job. if set to "Always", will always pull the latest image.
+                When "IfNotPresent", will only pull if no image has already been pulled.
+                Defaults to "IfNotPresent".
+        """
+        if jobname is None:
+            self.jobname = uuid.uuid4().hex
+        else:
+            self.jobname = jobname
+        if google_cloud_project is None:
+            try:
+                self.google_cloud_project = filesystem._get_gcloud_project()
+            except ImportError:
+                self.google_cloud_project = None
+        else:
+            self.google_cloud_project = google_cloud_project
+        self.memory_gb = memory_gb
+        if memory_gb_limit is None:
+            self.memory_gb_limit = memory_gb
+        else:
+            self.memory_gb_limit = memory_gb_limit
+        self.cpu_count = cpu_count
+        self.gcp_secret = gcp_secret
+        self.image_pull_policy = image_pull_policy
+
+    @property
+    def resource_requirements(self):
+        return kube.client.V1ResourceRequirements(
+            limits={
+                'memory': f'{self.memory_gb_limit:.1f}G',
+            },
+            requests={
+                'memory': f'{self.memory_gb:.1f}G',
+                'cpu': f'{self.cpu_count:.1f}'
+            },
+        )
+
+    @property
+    def _secret_volume(self):
+        if self.gcp_secret is not None:
+            return kube.client.V1Volume(
+                name='gcp-key-secret',
+                secret=kube.client.V1SecretVolumeSource(
+                    secret_name=self.gcp_secret
+                )
+            )
+        else:
+            return None
+    
+    @property
+    def volume_list(self):
+        if self.gcp_secret is not None:
+            return [self._secret_volume]
+        else:
+            return []
+    
+    @property
+    def volume_mounts_list(self):
+        if self.gcp_secret is not None:
+            volume_mounts_list = [
+                kube.client.V1VolumeMount(
+                    mount_path='/secret/gcp-credentials',
+                    name=self._secret_volume.name,
+                    read_only=True,
+                )
+            ]
+        else:
+            volume_mounts_list = []
+        return volume_mounts_list
+
+    @property
+    def _secret_env_list(self):
```

Comment:
`env` isn't 100% clear. I would say `_secret_environmental_variables`

Reply:
I used `env` because that is the name used by the kubernetes API.